client/components/editor.vue

```diff
@@ -358,52 +358,37 @@
           this.unconfirmedCount = this.$editorView.state.collab$.unconfirmed.length;
 
           const sendable = collab.sendableSteps(newState);
-          let addingComment = false;
-          let deletingComment = false;
-
           if (sendable) {
-            if (this.commentHighlightKey) {
-              const commentMarks = _.filter(transaction.steps, (s) => {
-                return s.mark && s.mark.type.name === 'highlight' && s.mark.attrs['highlight-keys'] === this.commentHighlightKey;
+            const containsHighlightStep = sendable.steps.find((x) => {
+              return (x.jsonID === 'addMark' || x.jsonID === 'removeMark') && x.mark && x.mark.type.name === 'highlight';
+            });
+            if (this.canUserUpdateDocument || (this.canUserCreateComments && (containsHighlightStep))) {
+              this.lastSentVersion = sendable.version;
+              this.addingStepsInProgress = true;
+              Content.addSteps({
+                contentKey: this.contentKey,
+                currentVersion: sendable.version,
+                steps: sendable.steps.map((step) => {
+                  return step.toJSON();
+                }),
+                clientId: this.clientId,
+              }, (error, response) => {
+                this.addingStepsInProgress = false;
+                if (!error) {
+                  response.highlightSteps.forEach((x) => {
```

Comment:
So why are you then processing the method return value? I thought we agreed that you will be observing what is coming confirmed from the server from pub/sub?

Reply:
Done, now the confirmed steps are observed in pub/sub.